libraries/AP_Compass/AP_Compass.cpp

```diff
@@ -938,6 +938,8 @@ void Compass::_detect_backends(void)
     ADD_BACKEND(DRIVER_LSM9DS1, AP_Compass_LSM9DS1::probe(hal.spi->get_device("lsm9ds1_m")));
 #elif HAL_COMPASS_DEFAULT == HAL_COMPASS_BMM150_I2C
     ADD_BACKEND(DRIVER_BMM150, AP_Compass_BMM150::probe(GET_I2C_DEVICE(HAL_COMPASS_BMM150_I2C_BUS, HAL_COMPASS_BMM150_I2C_ADDR)));
+#elif CONFIG_HAL_BOARD_SUBTYPE == HAL_BOARD_SUBTYPE_CHIBIOS_MROCONTROLZEROF7
```

Comment:
Why do you use board_subtype and not HAL_COMPASS_DEFAULT?

Reply:
@sh83 The sensor orientation is specific to this board so why not use CONFIG_HAL_BOARD_SUBTYPE

Reply:
also, HAL_COMPASS_DEFAULT doesn't allow for multiple sensors

Reply:
Orientation is also could set by define. See HAL_COMPASS_HMC5843_ROTATION